portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -32,43 +38,61 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  private List<Comment> commentsList;
-
-  @Override
-  public void init() {
-    commentsList = new ArrayList<>();
-    commentsList.add(Comment.create("Alice", "Wow, this website is great!"));
-    commentsList.add(Comment.create("Bob", "Cool website!"));
-  }
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Comment> comments = new ArrayList<>();
+    final String PROPERTY_NAME = "name";
+    final String PROPERTY_TEXT = "text";
+    final String PROPERTY_TIMESTAMP = "timestamp";
```

Comment:
Sorry, I meant to say, put them as the shared variables inside this class(And remember to define it as `private static final`). In this way, doPost can reuse the variable too.

One benefit is that, in future, you only need to change one place if you update your datastore schema.

Reply:
Ah, my bad for overlooking that. Just fixed it.